tests/mve/gtest_scene.cc

```diff
@@ -0,0 +1,751 @@
+#include <gtest/gtest.h>
+
+#include <cstdio>
+#include <iomanip>
+#include <sstream>
+#include <string>
+
+#include "mve/bundle_io.h"
+#include "mve/scene.h"
+#include "util/exception.h"
+#include "util/file_system.h"
+
+namespace {
+
+/* Cleanup mechanism. Unlinks files on destruction. */
+class OnScopeExit final
+{
+public:
+    ~OnScopeExit();
+    void unlink(std::string str);
+
+private:
+    void unlink_recursive(const std::string& str);
+
+    std::vector<std::string> paths;
+};
+
+/**
+ * Creates a valid scene directory in the system's temporary directory.
+ * This directory has an empty "views" subdirectory and
+ * a bundle file with zero cameras and zero features.
+ * @param view_count Number of views the scene is created with.
+ * @param bundle Bundle to include within the scene or
+ *     nullptr to create a scene without a bundle file.
+ * @return The path to the created scene directory.
+ */
+std::string create_scene_on_disk(const std::size_t view_count,
+                                 const mve::Bundle::Ptr bundle,
+                                 OnScopeExit& clean_up);
+
+void make_dirty(mve::View::Ptr view);
+void make_a_clean_view_dirty(mve::Scene::Ptr scene);
+mve::Scene::ViewList
+        load_views_directly_from(const std::string& scene_directory);
+mve::Bundle::Ptr
+        load_bundle_directly_from(const std::string& scene_directory);
+mve::Bundle::Ptr make_bundle(const std::size_t camera_count);
+bool views_match(mve::Scene::ViewList& lhs, mve::Scene::ViewList& rhs);
+bool bundle_cameras_match(mve::Bundle::Ptr lhs, mve::Bundle::ConstPtr rhs);
+
+} // anonymous namespace
+
+//== Test the initial state of a created scene =================================
+
+TEST(SceneTest,
+     a_created_scene_is_initialy_clean)
+{
+    OnScopeExit clean_up;
+
+    std::string scene_path = create_scene_on_disk(0, nullptr, clean_up);
+    mve::Scene::Ptr scene = mve::Scene::create(scene_path);
+    ASSERT_TRUE(scene != nullptr);
+    EXPECT_FALSE(scene->is_dirty());
+}
+
+TEST(SceneTest,
+     the_initial_path_of_a_created_scene_is_the_path_it_was_created_with)
```

Comment:
Put the test names on the same line as TEST. Use CamelCase for the test names like in the other tests. If the test name doesn't fit, just make the name a bit shorter :)